src/vivarium_census_prl_synth_pop/components/observers.py

```diff
@@ -124,21 +130,34 @@ def get_responses(self) -> pd.DataFrame:
             "sex",
             "race_ethnicity",
             "census_year",
+            "guardian_1",
+            "guardian_1_address",
+            "guardian_2",
+            "guardian_2_address",
+            "housing_type",
         ])
 
     def to_observe(self, event: Event) -> bool:
-        if (self.clock().year % 10 == 0) & (self.clock().month == 4):
-            if self.clock().day < 29:  # because we only want one observation in April  FIXME: cooler to do this with the timestep
-                return True
-        return False
-
+        return ((self.clock().year % 10 == 0)  # decennial year
+                and (self.clock().month == 4)  # month of April
+                and (self.clock().day <= self.time_step)  # time step containing first day of month
+                      # with current model spec, on first time step, event.time equals 2020-04-29
+                      # but self.clock() equals 2020-04-01
+               )
     def do_observation(self, event) -> None:
         pop = self.population_view.get(
-            event.index, # query="alive == 'alive'",  # TODO: uncomment this to include only living simulants in census
+            event.index,
+            query="alive == 'alive'",  # census should include only living simulants
         )
         pop["middle_initial"] = pop["middle_name"].astype(str).str[0]
         pop = pop.drop(columns="middle_name")
 
-        # TODO: include additional columns specified in MIC-3642
+        # merge addresses for guardian_1 and guardian_2 for the rows with guardians
+        for i in [1,2]:
+            s_guardian_id = pop[f"guardian_{i}"].dropna()
+            s_guardian_id = s_guardian_id[s_guardian_id != -1] # is it faster to remove the negative values?
+            pop[f"guardian_{i}_address"] = s_guardian_id.map(pop.address)
```

Comment:
This is going to be address_id and not address.  I do think we have had concerns about the performance on map.  This is also going to add a bunch of NaNs but I think that is ok.

Reply:
Let's keep an eye on runtime, here is what I found with the current model spec:
```
time simulate run --pdb -v src/vivarium_census_prl_synth_pop/model_specifications/model_spec.yaml
...
2022-12-15 15:36:24.448 | vivarium.framework.engine:report:203 -
              Event  Count  Mean time (s)  Std. dev. time (s)  Total time (s)  % Total time
     initialization      1           0.76                0.00            0.76          0.15
              setup      1          76.50                0.00           76.50         15.23
         post_setup      1           0.00                0.00            0.00          0.00
population_creation      1         176.82                0.00          176.82         35.21
 time_step__prepare    132           0.00                0.00            0.06          0.01
          time_step    132           1.87                0.27          246.66         49.11
 time_step__cleanup    132           0.01                0.00            1.24          0.25
    collect_metrics    132           0.00                0.00            0.13          0.03
     simulation_end      1           0.04                0.00            0.04          0.01
              total      1         502.20                0.00          502.20        100.00
```
I guess that `collect_metrics` is the line where this compute happens.  It is much less than 1% of the total run time currently.

Reply:
Speaking of runtime, I think Jim's work on addresses has sped up the sim by 40%.  Here is what I got when I tested against the new `main`:

```
2022-12-15 16:24:20.599 | vivarium.framework.engine:report:203 -
              Event  Count  Mean time (s)  Std. dev. time (s)  Total time (s)  % Total time
     initialization      1           0.68                0.00            0.68          0.22
              setup      1          76.25                0.00           76.25         25.12
         post_setup      1           0.00                0.00            0.00          0.00
population_creation      1         140.52                0.00          140.52         46.28
 time_step__prepare    132           0.00                0.00            0.06          0.02
          time_step    132           0.64                0.14           84.78         27.92
 time_step__cleanup    132           0.01                0.00            1.16          0.38
    collect_metrics    132           0.00                0.00            0.11          0.04
     simulation_end      1           0.04                0.00            0.04          0.01
              total      1         303.60                0.00          303.60        100.00
```